src/manager.rs

```diff
@@ -823,3 +823,19 @@ impl<C: ConfigStore> Manager<C, Registered> {
         Ok(service_cipher)
     }
 }
+
+impl<C> Manager<C, Registered>
+where
+    C: ConfigStore + MessageStore,
+{
+    pub async fn receive_messages_store(&self) -> Result<impl Stream<Item = Content> + '_, Error> {
+        let mut store = self.config_store.clone();
+        Ok(self.receive_messages().await?.map(move |c| {
+            if c.metadata.sender.uuid.is_some() {
+                // TODO: Error handling?
+                let _ = store.save_message(c.clone(), None::<ServiceAddress>);
```

Comment:
I think here, you could just log errors, I don't feel like we should interrupt the stream if we can't decode a single message.